api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/standalone/StandaloneLoaderService.java

```diff
@@ -0,0 +1,129 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.zowe.apiml.apicatalog.instance.InstanceInitializeService;
+import org.zowe.apiml.apicatalog.services.cached.CachedApiDocService;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+
+@Slf4j
+@Service
+@ConditionalOnProperty(
+    value = "apiml.catalog.standalone.enabled",
+    havingValue = "true")
+@RequiredArgsConstructor
+public class StandaloneLoaderService {
+
+    @Value("${apiml.catalog.standalone.servicesDirectory:services}")
+    private String servicesDirectory;
+
+    private final ObjectMapper objectMapper;
+    private final InstanceInitializeService instanceInitializeService;
+    private final CachedApiDocService cachedApiDocService;
+    private final StandaloneAPIDocRetrievalService standaloneAPIDocRetrievalService;
+
+    public void initializeCache() {
+        loadApplicationCache();
+        loadOpenAPICache();
+    }
+
+    private void loadApplicationCache() {
+        File[] appFiles = getFiles(servicesDirectory + "/apps");
+        if (appFiles.length == 0) {
+            log.error("No service definition files found.");
+            return;
+        }
+
+        for (File file : appFiles) {
+            createContainerFromFile(file);
+        }
+    }
+
+    private void loadOpenAPICache() {
+        File[] openAPIFiles = getFiles(servicesDirectory + "/apiDocs");
+        if (openAPIFiles.length == 0) {
+            log.error("No apiDocs files found.");
+            return;
+        }
+
+        for (File openAPIFile : openAPIFiles) {
+            loadApiDocCache(openAPIFile);
+        }
+    }
+
+    private void createContainerFromFile(File file) {
+        log.info("Initialising services from '{}' file.", file.getName());
+
+        try {
+            Applications apps = objectMapper.readValue(file, Applications.class);
+            apps.getRegisteredApplications().forEach(app -> {
+                instanceInitializeService.createContainers(app);
+                // Uses metadata from the first instance like {@link InstanceRetrievalService#getInstanceInfo}
+                loadApiDocVersionCache(app.getInstances().get(0));
+            });
+        } catch (IOException e) {
+            log.error("Unable to parse service definition '{}' because {}", file.getName(), e.getMessage());
+        }
+    }
+
+    private void loadApiDocCache(File file) {
+        try {
+            String[] name = FilenameUtils.removeExtension(file.getName()).split("_");
+            if (name.length < 2 || name.length > 3 || name.length == 3 && !"default".equals(name[2])) {
```

Comment:
```suggestion
            if (name.length < 2 || name.length > 3 || (name.length == 3 && !"default".equals(name[2]))) {
```
Just to be more readable.